test/gke/resize-cluster.sh

```diff
@@ -39,4 +39,15 @@ if [ "${#node_pools[@]}" -ne 1 ] ; then
   exit 1
 fi
 
-gcloud container clusters resize --project "${project}" --region "${region}" --node-pool "${node_pools[0]}" --num-nodes ${node_count} --quiet "${cluster_uri}"
+resize_retries=0
+scaled=1
+while [ $resize_retries -lt 4 ]; do
+	if gcloud container clusters resize --project "${project}" --region "${region}" --node-pool "${node_pools[0]}" --num-nodes "${node_count}" --quiet "${cluster_uri}" ; then
+		scaled=0
+		break
+	fi
+	sleep 15
+	((resize_retries++))
+done
+
+exit $scaled
```

Comment:
I was thinking about that last week, but is there any reason most of the pipelines' work is done in `.sh` and not in the pipelines themselves?

If not, I think it could be good to think about moving intelligence out of the scripts and into Jenkinsfiles when appropriate. For example in this case we could have used Jenkins retry blocks rather than a custom bash retry, which I feel is more flexible.

What do you think?

Reply:
I would say that the reason for that is that most people contributing are much more comfortable with bash than with jenkinsfiles.

We can definitely talk about moving all this logic to jenkinsfiles, but this script is called from within another script that is already wrapped in retries.

After cluster is resized there are still some other things to be done, so we would need to have yet another script in gke dir (which might be a good idea).